crates/api/api_utils/src/notify.rs

```diff
@@ -0,0 +1,814 @@
+    let jessica_post = Post::create(pool, &jessica_post_form).await?;
+
+    let timmy_comment_form =
+      CommentInsertForm::new(timmy.person.id, timmy_post.id, "timmy comment prv".into());
+    let timmy_comment = Comment::create(pool, &timmy_comment_form, None).await?;
+
+    let sara_comment_form =
+      CommentInsertForm::new(sara.person.id, timmy_post.id, "sara comment prv".into());
+    let sara_comment = Comment::create(pool, &sara_comment_form, Some(&timmy_comment.path)).await?;
+
+    Ok(Data {
+      instance,
+      timmy,
+      sara,
+      jessica,
+      community,
+      timmy_post,
+      jessica_post,
+      timmy_comment,
+      sara_comment,
+    })
+  }
+
+  async fn insert_private_message(
+    form: PrivateMessageInsertForm,
+    context: &LemmyContext,
+  ) -> LemmyResult<()> {
+    let pool = &mut context.pool();
+    let pm = PrivateMessage::create(pool, &form).await?;
+    let view = PrivateMessageView::read(pool, pm.id).await?;
+    notify_private_message(&view, false, context).await?;
+    Ok(())
+  }
+  async fn setup_private_messages(data: &Data, context: &LemmyContext) -> LemmyResult<()> {
+    let sara_timmy_message_form = PrivateMessageInsertForm::new(
+      data.sara.person.id,
+      data.timmy.person.id,
+      "sara to timmy".into(),
+    );
+    insert_private_message(sara_timmy_message_form, context).await?;
+
+    let sara_jessica_message_form = PrivateMessageInsertForm::new(
+      data.sara.person.id,
+      data.jessica.id,
+      "sara to jessica".into(),
+    );
+    insert_private_message(sara_jessica_message_form, context).await?;
+
+    let timmy_sara_message_form = PrivateMessageInsertForm::new(
+      data.timmy.person.id,
+      data.sara.person.id,
+      "timmy to sara".into(),
+    );
+    insert_private_message(timmy_sara_message_form, context).await?;
+
+    let jessica_timmy_message_form = PrivateMessageInsertForm::new(
+      data.jessica.id,
+      data.timmy.person.id,
+      "jessica to timmy".into(),
+    );
+    insert_private_message(jessica_timmy_message_form, context).await?;
+
+    Ok(())
+  }
+
+  async fn cleanup(data: Data, pool: &mut DbPool<'_>) -> LemmyResult<()> {
+    Instance::delete(pool, data.instance.id).await?;
+
+    Ok(())
+  }
+
+  #[tokio::test]
+  #[serial]
+  async fn replies() -> LemmyResult<()> {
+    let context = LemmyContext::init_test_context().await;
+    let pool = &mut context.pool();
+    let data = init_data(pool).await?;
+
+    // Sara replied to timmys comment, but lets create the row now
+    NotifyData {
+      post: &data.timmy_post,
+      comment_opt: Some(&data.sara_comment),
+      creator: &data.sara.person,
+      community: &data.community,
+      do_send_email: false,
+    }
+    .send(&context)
+    .await?;
+
+    let timmy_unread_replies =
+      NotificationView::get_unread_count(pool, data.timmy.person.id, data.instance.id, true)
+        .await?;
+    assert_eq!(1, timmy_unread_replies);
```

Comment:
This is failing because the same notification is returned twice. The actual data in the db is correct, but both count and list return wrong data. Seems to be something wrong with notification joins but I cant figure out what.

Reply:
Is this still an issue?

Reply:
Now this fixed itself.